src/config.rs

```diff
@@ -284,6 +287,83 @@ impl GlobalConfig {
             ColorChoice::Never | ColorChoice::Auto => false,
         }
     }
+
+    /// Set (overwrite) the [`FeatureFlag`] set.
+    #[inline]
+    pub fn set_feature_flags(&mut self, flags: HashSet<FeatureFlag>) -> &mut Self {
+        self.feature_flags = flags;
+        self
+    }
+
+    /// Enable a single [feature flag](FeatureFlag).
+    #[inline]
+    pub fn enable_feature_flag(&mut self, flag: FeatureFlag) -> &mut Self {
+        self.feature_flags.insert(flag);
+        self
+    }
+
+    /// Test for whether a specific feature flag is enabled.
+    #[must_use]
+    #[inline]
+    pub fn feature_flag_enabled(&self, flag: FeatureFlag) -> bool {
+        self.feature_flags.contains(&flag)
+    }
+
+    /// Returns a set of all enabled feature flags.
+    #[must_use]
+    #[inline]
+    pub fn feature_flags(&self) -> &HashSet<FeatureFlag> {
+        &self.feature_flags
+    }
+}
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+pub struct FeatureFlag {
+    /// `kebab-case` identifier for this feature flag.
+    pub id: &'static str,
+    /// Optional help message for this flag, separated by line breaks.  The first line
+    /// should be able to serve as a one-line short help message.
+    pub help: Option<&'static [&'static str]>,
```

Comment:
To make sure I understand this right, could you elaborate what the elements in the slice should be? The element at `[0]` is the one-line short help message, how about the rest? Do we ever expect more than 1-2 elements? How would they be printed?

Reply:
That's what I intended, I think `clap` does this behavior internally.  I changed it to print the whole long help message (formatted by lines) with `-Z help`, but we do provide the `help` string to `clap` (although it's currently hidden), so I added a doc-comment that the first line should be able to be used as a short help message.

Reply:
I'm just not sure how the remaining elements in the slice get printed. I wouldn't know whether I'm expected to offer exactly two elements here, or offer one element per line, or one element per paragraph.

Could you elaborate the doc comment on those points, for the benefit of future code reviewers' ability to review `help` text for any future options being added here.

Reply:
Ah I see you've already changed it :+1: